src/components/Canvas.vue

```diff
@@ -1,40 +1,55 @@
 <template>
-  <div style="position: relative">
-    <p-toolbox :types="nodeTypes" @add="addNode" />
-    <p-background>
-      <p-edge
-        v-if="newEdge"
-        ref="newEdge"
-        :start="newEdge.start.$refs ? newEdge.start.$refs.connector : newEdge.start"
-        :end="newEdge.end.$refs ? newEdge.end.$refs.connector : newEdge.end"
-        :color="newEdge.color"
-      />
-      <p-edge
-        v-for="edge in edges"
-        :ref="edge.id"
-        :key="edge.id"
-        :start="edge.start.$refs.connector"
-        :end="edge.end.$refs.connector"
-        :color="edge.color"
-      />
-      <p-node
-        v-for="node in nodes"
-        :key="node.id"
-        :instance="node"
-        :x="300"
-        :y="100"
-        ref="nodes"
-        @connect="onConnect"
-        @move="onNodeMove"
-      >
-        <component :is="node.component" />
-      </p-node>
+  <div id="canvas">
+    <p-toolbox id="toolbox" :types="nodeTypes" @add="addNode" />
+    <p-background v-touch-pan.mouse.prevent="handlePan" :scroll="scroll">
+      <div :style="{position: 'absolute', top: -scroll.y + 'px', left: -scroll.x + 'px'}">
```

Comment:
Inserting spaces to seperate the object curly braces might be more consistent in terms of codestyle. Have you done it this way to be more compact?

Reply:
My bad. I thought our linter settings disallowed spaces around object parentheses.